src/app/dashboard/summary/summary.component.ts

```diff
@@ -107,19 +214,39 @@ export class SummaryComponent implements OnChanges {
         ).format(value)
     }
 
-    changeSelectedSummaryComponent(e: any) {
+    changeSelectedSummaryComponent(e: MouseEvent) {
         // if nodeName is APP-BIG-NUMBER our actual target is a child - thus not findable with .closest
-        const newSelected = e.target.nodeName != "APP-BIG-NUMBER" ? e.target.closest(".big_number") : e.target.children[0].closest(".big_number")
-        const siblings = [...newSelected.parentNode.parentNode.children];
-        siblings.forEach((e) => e.children[0].children[0].classList.remove("selected"))
-        if(newSelected.children)
-            newSelected.children[0].classList.add("selected")
+        const target = e.target as HTMLElement;
+        if (!target) return;
+
+        const newSelected = target.nodeName !== "APP-BIG-NUMBER"
+            ? target.closest(".big_number") as HTMLElement
+            : (target.children[0] as HTMLElement)?.closest(".big_number") as HTMLElement;
+
+        if (!newSelected) return;
+
+        const parentNode = newSelected.parentNode?.parentNode;
+        if (!parentNode) return;
+
+        const siblings = Array.from(newSelected.parentNode!.parentNode!.children) as HTMLElement[];
+        siblings.forEach((element) => {
+            const firstChild = element.children[0] as HTMLElement;
+            const secondChild = firstChild?.children[0] as HTMLElement;
+            secondChild?.classList.remove("selected");
+        });
+
+        const firstChild = newSelected.children[0] as HTMLElement;
+        firstChild?.classList.add("selected");
```

Comment:
why was this change needed? Did the behavior change? I feel like it looks more complicated now than before - but the before does not really look clean either.. Maybe there is a cleaner way which does not rely on html selectors (parentnode, children, etc) at all?